common/ffsparser.cpp

```diff
@@ -3366,156 +3379,233 @@ USTATUS FfsParser::checkProtectedRanges(const UModelIndex & index)
     UByteArray protectedParts;
     bool bgProtectedRangeFound = false;
     try {
-        for (UINT32 i = 0; i < (UINT32)bgProtectedRanges.size(); i++) {
-            if (bgProtectedRanges[i].Type == BG_PROTECTED_RANGE_INTEL_BOOT_GUARD_IBB && bgProtectedRanges[i].Size > 0) {
+        for (UINT32 i = 0; i < (UINT32)protectedRanges.size(); i++) {
+            if (protectedRanges[i].Type == PROTECTED_RANGE_INTEL_BOOT_GUARD_IBB && protectedRanges[i].Size > 0) {
                 bgProtectedRangeFound = true;
-                if ((UINT64)bgProtectedRanges[i].Offset >= addressDiff) {
-                    bgProtectedRanges[i].Offset -= (UINT32)addressDiff;
+                if ((UINT64)protectedRanges[i].Offset >= addressDiff) {
+                    protectedRanges[i].Offset -= (UINT32)addressDiff;
                 } else {
-                    // TODO: Explore this.
-                    msg(usprintf("%s: suspicious BG protection offset", __FUNCTION__), index);
+                    msg(usprintf("%s: suspicious protected range offset", __FUNCTION__), index);
                 }
-                protectedParts += openedImage.mid(bgProtectedRanges[i].Offset, bgProtectedRanges[i].Size);
-                markProtectedRangeRecursive(index, bgProtectedRanges[i]);
+                protectedParts += openedImage.mid(protectedRanges[i].Offset, protectedRanges[i].Size);
+                markProtectedRangeRecursive(index, protectedRanges[i]);
             }
         }
     } catch (...) {
         bgProtectedRangeFound = false;
     }
     
     if (bgProtectedRangeFound) {
-        UByteArray digest(SHA256_DIGEST_SIZE, '\x00');
-        sha256(protectedParts.constData(), protectedParts.size(), digest.data());
+        UINT8 digest[SHA512_HASH_SIZE] = {};
+        UString digestString;
+        UString ibbDigests;
+        // SHA1
```

Comment:
Maybe put that to a function and then call with hash pointer, ascii name and digest size? Feels a bit too copy-pasted.

Reply:
Yep, there's are lots of such in fitparser.cpp also.